packages/vite-node/src/server.ts

```diff
@@ -11,19 +11,33 @@ import { withInlineSourcemap } from './source-map'
 
 export * from './externalize'
 
+interface FetchCache {
+  duration?: number
+  timestamp: number
+  result: FetchResult
+}
+
 const debugRequest = createDebug('vite-node:server:request')
 
 export class ViteNodeServer {
-  private fetchPromiseMap = new Map<string, Promise<FetchResult>>()
-  private transformPromiseMap = new Map<string, Promise<TransformResult | null | undefined>>()
+  private fetchPromiseMap = {
+    ssr: new Map<string, Promise<FetchResult>>(),
+    web: new Map<string, Promise<FetchResult>>(),
+  }
+
+  private transformPromiseMap = {
+    ssr: new Map<string, Promise<TransformResult | null | undefined>>(),
+    web: new Map<string, Promise<TransformResult | null | undefined>>(),
+  }
 
   private existingOptimizedDeps = new Set<string>()
 
-  fetchCache = new Map<string, {
-    duration?: number
-    timestamp: number
-    result: FetchResult
-  }>()
+  fetchCaches = {
+    ssr: new Map<string, FetchCache>(),
+    web: new Map<string, FetchCache>(),
+  }
+
+  fetchCache = new Map<string, FetchCache>()
```

Comment:
Is this one left here for backwards compatibility or why is it needed?

Reply:
Yes